tracer/src/Datadog.Trace/Iast/SensitiveData/JsonTokenizer.cs

```diff
@@ -0,0 +1,85 @@
+// <copyright file="JsonTokenizer.cs" company="Datadog">
+// Unless explicitly stated otherwise all files in this repository are licensed under the Apache 2 License.
+// This product includes software developed at Datadog (https://www.datadoghq.com/). Copyright 2017 Datadog, Inc.
+// </copyright>
+
+using System.Collections.Generic;
+using System.IO;
+using System.Text.RegularExpressions;
+using Datadog.Trace.Configuration;
+using Datadog.Trace.Vendors.Newtonsoft.Json;
+
+#nullable enable
+
+namespace Datadog.Trace.Iast.SensitiveData;
+
+internal class JsonTokenizer : ITokenizer
+{
+    private static readonly Regex SourceValueRegex = new(@"(?i)bearer\s+[a-z0-9\._\-]+|token:[a-z0-9]{13}|gh[opsu]_[0-9a-zA-Z]{36}|ey[I-L][\w=-]+\.ey[I-L][\w=-]+(\.[\w.+\/=-]+)?|[\-]{5}BEGIN[a-z\s]+PRIVATE\sKEY[\-]{5}[^\-]+[\-]{5}END[a-z\s]+PRIVATE\sKEY|ssh-rsa\s*[a-z0-9\/\.+]{100,}", RegexOptions.Compiled);
```

Comment:
We recently added timeouts in the constructors of all the tokenizers to avoid DOS attacks. It's something very recent. Basically, you would need to add a constructor with the timeout. You can find examples here: https://github.com/DataDog/dd-trace-dotnet/pull/5083/files